### ISSUE TITLE
Recurring Error while Synching Beaon with Lodestar

### ISSUE THREAD

**Issue by jeyakatsa** [ChainSafe/lodestar#3220]
Been getting this Error (recurring) while synching Beacon with Lodestar to mainnet:

```
Error processing finalized checkpoi nt epoch=11891 message=No block found for slot 380416 root 0xcc41812983f695363ac326491d51a98c3856ec9c2b60489314c2aa3322478348

Error: No block found for slot 380416 root 0xcc41812983f695363ac326491d51a98c3856ec9c2b60489314c2aa3322478348
    at C:\Users\Jeyak\AppData\Roaming\npm\node_modules\@chainsafe\lodestar-cli\node_modules\@chainsafe\lodestar\src\chain\archiver\archiveBlocks.ts:46:19
    at async Promise.all (index 94)
    at archiveBlocks (C:\Users\Jeyak\AppData\Roaming\npm\node_modules\@chainsafe
\lodestar-cli\node_modules\@chainsafe\lodestar\src\chain\archiver\archiveBlocks.
ts:40:7)
    at JobItemQueue.Archiver.processFinalizedCheckpoint [as ItemProcessor] (C:\Users\Jeyak\AppData\Roaming\npm\node_modules\@chainsafe\lodestar-cli\node_modules\@chainsafe\lodestar\src\chain\archiver\index.ts:71:7)
    at Timeout.JobItemQueue.runJob [as onTimeout] (C:\Users\Jeyak\AppData\Roaming\npm\node_modules\@chainsafe\lodestar-cli\node_modules\@chainsafe\lodestar\src\util\queue\itemQueue.ts:85:22)
```

**Comment by dapplion** [comment]
Hi @jeyakatsa ! Thank you so much for reporting :heart: What version of Lodestar are you running (paste `lodestar --version`)? Could you also paste the full command used to run it?

**Comment by jeyakatsa** [comment]
Hi @dapplion this is the full command used to run beacon: `lodestar beacon --network`. I'm using GitBash terminal to execute Beacon on Windows 10.

Will inform you which version of Lodestar I'm running right after the synch is complete (if I try to paste said request via the terminal right now, it'll cancel the synch and I'll have to start it again (which takes hours/days to complete)). I'll get back to you with that info as soon as I can. 

Thanks.

**Comment by jeyakatsa** [comment]
Hi @dapplion here is the Lodestar version that I'm running:

```
🌟 Lodestar: TypeScript Implementation of the Ethereum 2.0 Beacon Chain.
  * Version: v0.30.0 (npm)
  * by ChainSafe Systems, 2018-2021
```

Let me know if you need anything else.

**Comment by philknows** [comment]
Hi @jeyakatsa, are you still getting a similar issue with any of the subsequent versions we've published? Are you able to try the latest `v0.33.0` and let us know if this is still reproducible? Thanks!

**Comment by dapplion** [comment]
Closing since this part of the code has been refactored. Please re-open if it happens again!
